### DISCUSSION TITLE
Local prompt files for dev

### DISCUSSION THREAD

**Discussion by debkanchan** [langfuse/langfuse#4550] [Ideas]
### Describe the feature or potential improvement

Langfuse prompt management is great! But what if I want to tinker with the prompt locally during dev?

Current way to do this is to either remove the current prompt and add my prompt manually or to wrap the prompt management in another function that loads the prompt according to the env.

I'd like to propose an alternative way inspired by [Firebase GenKit](https://firebase.google.com/docs/genkit/dotprompt):
- For each prompt `prompt-name` in Langfuse we can have `prompt-name.prompt.txt` either at a specific location like project root or preferably some configurable location. `const prompt = await langfuse.getPrompt("jokes", {local="./prompts})`
- At runtime Langfuse would check if the runtime env is production or dev. If the env is dev then load the prompt from the local file if it exists.
- if the local prompt file does not exist it proceeds to load the prompt from Langfuse.

**Comment by marcklingen** [comment]
Appreciate that you share this here after mentioning this during the community session on Discord. Let's track interest for this here!

**Comment by cryoff** [comment]
I had exactly same requirements, and it was just easier to build it locally small-scale and re-use Langfuse data model : )
